autopush/websocket.py

```diff
@@ -327,9 +330,27 @@ def _check_router(self, paused=False, bridge_register=None):
             self.sendMessage(json.dumps(msg).encode('utf8'), False)
             return
         # User exists?
+        # Attempt to register the user for this session
+        # First, check to see if this uaid is registered somewhere else
+        record = self.ap_settings.router.get_uaid(self.uaid)
+        if record:
+            killUrl = record.get("node_id")
+            # Only delete foreign registers. We can handle the others.
+            if killUrl != self.ap_settings.router_url:
+                log.msg("Killing duplicate uaid: %s at %s" %
+                        (self.uaid, killUrl))
+                killUrl = killUrl + "/notif/" + self.uaid
+                d = self.ap_settings.agent.request(
+                    "DELETE",
+                    killUrl.encode("utf8"),
+                ).addCallback(self._reg_user)
+                d.addErrback(self.log_err, extra="Failed to kill old client")
+                return
```

Comment:
So, I don't think we should have connection nodes attempt to nuke dupe clients, for two reasons:

- Any client that is disconnected, then has a notification come in, will have its node entry deleted. So this only helps a wee bit.
- There's some concurrency issues having them try to call back and forth to delete entries, they'll need to include the connect time to ensure they don't accidentally tell a different node to delete a possibly new and active client that has reconnected in the span of this processing.

If the websocket ping frame is low enough (3 mins), and we delete dupe connections on the same node, that should keep connection churn low enough as is.